Ricochet/Assets/_Scripts/Managers/GameManager.cs

```diff
@@ -342,22 +382,45 @@ public void BallPlayerCollision(GameObject player, Collision2D collision)
         {
             if (gameMode == EMode.Deathmatch)
             {
-                if (modeManager.AltUpdateScore(playerController.GetTeamNumber(), -1))
+                if (playerController.GetTeamNumber() == ETeam.BlueTeam)
                 {
-                    if (playerController.GetTeamNumber() == ETeam.BlueTeam)
+                    if (modeManager.AltUpdateScore(ETeam.RedTeam, 1))
                     {
-                        gameData.SetGameWinner(ETeam.RedTeam);
-                        EndMatch();
+                        if (playerController.GetTeamNumber() == ETeam.BlueTeam)
```

Comment:
There's a lot of duplicated code and extra If statements here. Try to clean it up and make it more efficient.

Reply:
A bit of that isnt mine, but I will!